### DISCUSSION TITLE
How would you access/modify the avatar distance model in the code? (Attempting to change it via a custom chat command)

### DISCUSSION THREAD

**Discussion by rishivinod7** [mozilla/hubs#3262] [General Questions & Discussion] [answered]
I'm editing the message-dispatch.js file and I'm trying to see how to change the distance model by chat command

**Comment by robinkwilson** [comment, marked as answer]
Have you looked at how waypoints work in the code? Or moving the camera/the model to a specific x y z coordinates. -- not completely clear on your question, but hope something in there helps.

**Reply by rishivinod7** [reply]
I'm trying to make a command that could sort of toggle/switch between linear, exponential and inverse avatar distance models. This param seems to be set in the avatar-audio-source.js file from what I've found. I'm having trouble writing the code to make this possible. I would be happy enough to see if I can modify even the avatar max distance or ref distance.

**Reply by robinkwilson** [reply]
Here's a PR example of a chat command recently implemented as an example: https://github.com/mozilla/hubs/pull/2824

Other than that, I'm afraid this is a little too specific for us to answer completely in the discussion thread since we're a small team, we will leave this discussion thread open for a while to see if anyone else has any feedback!

I'd try discord (https://discord.gg/wHmY4nd), we don't guarantee an answer, but if our community members or one of the devs has time, they'll try to help you! Good luck!

**Reply by rishivinod7** [reply]
Ok, thank you for the response!
